control_file_gen.pl

```diff
@@ -610,17 +617,20 @@
    }
    #
    # fill in nodal attribute default values from hash
+   my $numLines = 0;
    while(<$nafi>) {
       foreach my $key (keys %{$p->{nodal_attributes}->{default_values}}) {
          my $tag = "%"."$key"."_default"."%";
          my $value = "$p->{nodal_attributes}->{default_values}->{$key}";
          s/$tag/$value/;
       }
       print $nafo $_;
+      $numLines++;
    }
```

Comment:
```suggestion
   # accumulate first 50 block of lines
   my $numLines = 0;
   my $first50lines;
   for my $line (1 .. 50) {
     $first50lines .= <$nafi>;
   }

   # global s///
   foreach my $key (keys %{$p->{nodal_attributes}->{default_values}}) {
     my $tag   = "%"."$key"."_default"."%";
     my $value = $p->{nodal_attributes}->{default_values}->{$key};
     $first50lines =~ s/$tag/$value/g;
   }

   print $nafo $first50lines;

   foreach my $line (<$nafi>) {
     print $nafo $line;
   }
```

Reply:
Thank you @wwlwpd I have now implemented this and tested it successfully.